netboxlabs/diode/sdk/client.py

```diff
@@ -295,10 +298,13 @@ def ingest(
         self,
         entities: Iterable[Entity | ingester_pb2.Entity | None],
         stream: str | None = _DEFAULT_STREAM,
+        *,
+        metadata: Metadata | None = None,
     ) -> ingester_pb2.IngestResponse:
-        """Ingest entities."""
+        """Ingest entities with optional request-level metadata."""
         for attempt in range(self._max_auth_retries):
             try:
+                request_metadata = convert_dict_to_struct(metadata) if metadata else None
```

Comment:
Nitpick. This block can be simplified as

```suggestion
                request = ingester_pb2.IngestRequest(
                    stream=stream,
                    id=str(uuid.uuid4()),
                    entities=entities,
                    sdk_name=self.name,
                    sdk_version=self.version,
                    producer_app_name=self.app_name,
                    producer_app_version=self.app_version,
                )
                if metadata is not None:
                    request_metadata = convert_dict_to_struct(metadata)
                    request.metadata.CopyFrom(request_metadata)
                return self._stub.Ingest(request, metadata=self._metadata)
```

to reduce the scope of request_metadata and reduce the number of conditions required

Reply:
Sadly the suggestion was incorrectly selected in first place, needs manual fix.

Reply:
Sadly the suggestion was incorrectly selected in first place, needs manual fix.